examples/autodiff.jl

```diff
@@ -0,0 +1,146 @@
+# call.
+Enzyme.autodiff(f, Duplicated(x, bx), Duplicated(y, by));
+# This yields the gradient of `f` in `bx` at point `x = [2.0, 2.0]`. `by` is called the seed and has
+# to be set to ``1.0`` in order to compute the gradient. Let's save the gradient for later.
+g = copy(bx)
+
+# # Forward mode
+# The forward model in AD is defined as
+# ```math
+# \begin{aligned}
+# y &= f(x) \\
+# \dot{y} &= \nabla f(x) \cdot x
+# \end{aligned}
+# ```
+# To obtain the first element of the gradient using the forward model we have to
+# seed ``\dot{x}`` with ``\dot{x} = [1.0,0.0]``
+x  = [2.0, 2.0]
+dx = [1.0, 0.0]
+y  = [0.0]
+dy = [0.0];
+# In the forward mode the second element of `Duplicated` stores the tangent.
+Enzyme.autodiff(Forward, f, Duplicated(x, dx), Duplicated(y, dy));
+
+# We can now verify that indeed the reverse mode and forward mode yield the same
+# result for the first component of the gradient. Note that to acquire the full
+# gradient one needs to execute the forward model a second time with the seed
+# `dx` set to `[0.0,1.0]`.
+
+# Let's verify whether the reverse and forward model agree.
+g[1] ≈ dy[1]
+
+# # Forward over reverse
+# The forward over reverse (FoR) model is obtained by applying the forward model
+# to the reverse model using the chain rule for the product in the adjoint statement.
+# ```math
+# \begin{aligned}
+# y &= f(x) \\
+# \dot{y} &= f(x) \cdot \dot{x} \\
+# \bar{x} &= \bar{y} \cdot \nabla f(x) \\
+# \dot{\bar{x}} &= \bar{y} \cdot \nabla^2 f(x) \cdot \dot{x} + \dot{\bar{y}} \cdot \nabla f(x)
+# \end{aligned}
+# ```
+# To obtain the first column/row of the Hessian ``\nabla^2 f(x)`` we have to
+# seed ``dot{\bar{y}}`` with ``[0.0]``, ``\bar{y}`` with ``[1.0]`` and ``\dot{x}`` with ``[1.0, 0.0]``.
+
+y = [0.0]
+x = [2.0, 2.0]
+
+dy = [0.0]
+dx = [1.0, 0.0]
+
+bx = [0.0, 0.0]
+by = [1.0]
+dbx = [0.0, 0.0]
+dby = [0.0]
+
+Enzyme.autodiff(
+    Forward,
+    (x,y) -> Enzyme.autodiff_deferred(f, x, y),
+    Duplicated(Duplicated(x, bx), Duplicated(dx, dbx)),
+    Duplicated(Duplicated(y, by), Duplicated(dy, dby)),
+)
+
+# The FoR model also computes the forward model from before, giving us again the first component of the gradient.
+g[1] ≈ dy[1]
+# In addition we now have the first row/column of the Hessian.
+dbx[1] ≈ 2.0
+dbx[2] ≈ 1.0
+
+# # Vector forward over reverse
+# The vector FoR allows us to propagate several tangents at once through the
+# second-order model. This allows us the acquire the Hessian in one autodiff
+# call. The multiple tangents are organized in tuples. Following the same seeding strategy as before, we now seed both
+# in the `vdx[1]=[1.0, 0.0]` and `vdx[2]=[0.0, 1.0]` direction. These tuples have to be put into a `BatchDuplicated` type.
+y = [0.0]
+x = [2.0, 2.0]
+
+vdy = ([0.0],[0.0])
+vdx = ([1.0, 0.0], [0.0, 1.0])
+
+bx = [0.0, 0.0]
+by = [1.0]
+vdbx = ([0.0, 0.0], [0.0, 0.0])
+vdby = ([0.0], [0.0]);
+
+# The `BatchedDuplicated`` objects are constructed using the broadcast operator
```

Comment:
The rendering is broken because of one additional tick:
```
`BatchedDuplicated`
```

Reply:
Thanks for catching!